### ISSUE TITLE
Ideas

### ISSUE THREAD

**Issue by LeoAlt8** [weidengeist/Willowbot#8]
I have a few more ideas for improvement.

1. When the connection is broken and then restored, the bot often does not connect to the chat (does not listen to it):
```
Reconnection trial 1 of 10 in 1 seconds …

Pass sent successfully
Bot account name xxxx sent successfully.
Entered channel xxxxxx.
```
It just closed a couple of times. But I'm not sure that this happened precisely because of the disconnect.

2. Add "listening" chat only when streaming online. This is so that the bot does not work 24/7 on the server. (If someone decides to upload it to a miniserver. To not write messages `"interval" :` when the stream is offline). Or someone wants to download it for free replit.

Maybe there are more ideas, I'll add them. Thank you very much for your work!!!

**Comment by weidengeist** [comment]
> 3. Add trigger types - bits (add a reaction as a sub).
> 
> 4. Add the ability to insert a timer (countdown) in response to a command.  For example, "There are xx hours xx minutes xx seconds left before the start of the event"

3. Unfortunately, bits are not possible to react to due to Willowbot’s design. Twitch does not handle bits via the IRC chat and therefore does not produce a message that could be processed by Willowbot.
4. This feature is beyond Willowbot’s intentional use case of simply handling and reacting to IRC messages. It’s not impossible, though. For such an individual function, it is recommended to [write your own module](https://github.com/weidengeist/Willowbot#4-optionalcustom-modules), which will be even less limited than Willowbot’s simple `if $message then $reaction` behaviour, as you can use the whole repertoire Python can offer.

**Comment by weidengeist** [comment]
> 5. Add installation method for VPS. :pray::pray::pray::pleading_face:

Asking for an installation method that is a highly specific and no general use case and which will have a tiny target group is like asking for the production of kitchen knives for people with two fingers and a thumb on their dominant hand. A VPS is not too special and any kind of software can be dropped there. Furthermore, there is no such thing as an installation for Willowbot. It is merely a collection of Python scripts working together. The only thing it needs is a Python installation. So your question should rather read: How to install Python on a VPS? This, however, is none of Willowbot’s business and is nothing that Willowbot is supposed or even able to do. (Without a Python installation on the target system it could not run anyway, so it would not be able to run itself to install Python to be able to run. You see the paradox.) Please read the documentation of VPS to learn how it works and how servers work in general. 

Long story short: An installation method for VPS is not going to be implemented.

**Comment by weidengeist** [comment]
1. [Added](https://github.com/weidengeist/Willowbot/commit/fa3bcb9b3b5edd604629522d69bc1b4fdd0c8263).
2. [Added](https://github.com/weidengeist/Willowbot/commit/fa3bcb9b3b5edd604629522d69bc1b4fdd0c8263#diff-43ecaecbf5c4f34a0341398c70caab306a85b0174f8d33420fbc7e96508ff936R265).
3. Discussed [here](https://github.com/weidengeist/Willowbot/issues/8#issuecomment-1458148659).
4. See 3.
5. Discussed [here](https://github.com/weidengeist/Willowbot/issues/8#issuecomment-1509006196).
6. Won’t be added. To specific for Willowbot’s intended general use case. Use Willowbot’s [custom modules](https://github.com/weidengeist/Willowbot?tab=readme-ov-file#4-optionalcustom-modules).

Discussion closed due to not being an issue tracker.
